### PR TITLE
Support outgoing message correction

### PR DESCRIPTION
**Pull request by raucao**
As per XEP-0308, and with the same AS object format as incoming corrections are forward to clients.

(Note: the XML fakes are still not tested correctly. I added FIXME comments, but since I wasn't able to drive-by-fix them easily, and they're already broken in master since before I worked on message corrections, I consider the problem out of scope for these feature PRs.)

### CODE REVIEW

**Comment by silverbucket** [comment]
@raucao Could you maybe create an issue and explain what you mean by wrong message? Do you mean any arbitrary data passes or that something specific about the type?

**Comment by raucao** [comment]
> Do you mean any arbitrary data passes

Yes.

File: packages/sockethub-platform-xmpp/src/index.test.js [hunk 1]
```diff
@@ -67,7 +67,17 @@ const job = {
...
+        content: 'hi all'
+      },
+      target: target.roomuser
+    },
+    correction: {
+      actor: actor,
+      object: {
+        'type': 'message',
+        id: 'hc-1234abcd',
+        replace: { id: 'hc-234bcde' },
```
Comment (raucao) [inline packages/sockethub-platform-xmpp/src/index.test.js]: Actually, I think this is `xmpp:replace` in the other direction.

image

Comment (raucao) [reply packages/sockethub-platform-xmpp/src/index.test.js]: Fixed.

**Comment by raucao** [comment]
I have fixed the property name. Should be good to merge now.

**Comment by raucao** [comment]
Can we merge this now?

**Review by silverbucket — APPROVED** [review]

**Comment by silverbucket** [comment]
lgtm!

**Pull request merged**